test/test_distributed.py

```diff
@@ -484,21 +551,150 @@ def _test_barrier_helper(self, group, group_id, rank):
 
         self._barrier()
 
+    @unittest.skipIf(BACKEND == 'nccl', "Nccl does not support CPU tensors")
     def test_barrier(self):
         group, group_id, rank = self._init_global_test()
         self._test_barrier_helper(group, group_id, rank)
 
+    @unittest.skipIf(BACKEND == 'nccl', "Nccl does not support newGroup")
     def test_barrier_group(self):
         group, group_id, rank = self._init_group_test()
         self._test_barrier_helper(group, group_id, rank)
 
-if BACKEND == 'tcp' or BACKEND == 'gloo':
+    def _init_multigpu_helper(self):
+        nGPUs = torch.cuda.device_count()
+        world_size = dist.get_world_size()
+        visible_devices = range(nGPUs)
+        nGPUs_per_process = int(nGPUs / world_size)
```

Comment:
nit: `nGPUs // world_size` will do the same thing. Also, should we ensure it's more than 0?

Reply:
Hi @apaszke , sorry for the confusion. The way I designed the multigpu testcases is 
- Spawn two processes( can be any number)
- Equally divide visible gpus to each process, I tested on a 8GPU machine, so this is 4GPU/process.
- This is to simulate the case when each process has multiple GPU and it's designed to test the NCCL *_multigpu APIs since those functions take a list of Tensors as input/output, each tensor resides on a different GPU. https://github.com/pytorch/pytorch/blob/master/torch/distributed/__init__.py#L253